### PR TITLE
Functionality to upload files

### PR DESCRIPTION
**Pull request by rafaelcdelgado**
- Create the scaffolding to add documents and refactor some code

Closes: https://github.com/Grupo-AFAL/documentation/issues/57

### CODE REVIEW

File: app/controllers/documentation/documents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Documentation
+  class DocumentsController < ApplicationController
+    include ActionView::RecordIdentifier
+    before_action :set_page
+
```
Comment (alexisKinto) [inline app/controllers/documentation/documents_controller.rb:7]:
```suggestion
    before_action :set_page
    before_action :set_documents

```

File: app/controllers/documentation/documents_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def destroy
+      @document = @page.documents.find(params[:id])
+      @document.destroy
+
+      render turbo_stream: turbo_stream.remove(dom_id(@document))
+    end
+
+    def set_page
+      @page = Page.find(params[:page_id])
+    end
```
Comment (alexisKinto) [inline app/controllers/documentation/documents_controller.rb:29]:
```suggestion
    def set_page
      @page = Page.find(params[:page_id])
    end
    
    def set_documents
      @documents = @page.documents
    end
```

File: app/models/documentation/page.rb [hunk 3]
```diff
@@ -45,5 +50,15 @@ def title_with_ancestors
...
     end
+
+    def file_extension(file)
+      file.blob.filename.to_s.split('.').last.upcase
+    end
+
+    def allowed_extensions(kind = nil)
+      return ALLOWED_DOCUMENT_EXTENSIONS.join(',') if kind == :document
+
+      ALLOWED_IMAGE_EXTENSIONS.join(',')
```
Comment (alexisKinto) [inline app/models/documentation/page.rb:61]: I think we should only return image extension on `kind == :image`, else return an empty array

File: app/views/documentation/documents/_list.html.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      </td>
+      <td>
+        <%= page.file_extension(document) %>
+      </td>
+      <td>
+        <%= l(document.created_at, format: :date_twelve_hours) %>
+      </td>
+      <td>
+        <%= link_to render(Bali::Icon::Component.new('trash')), page_document_path(@page, document),
+              class: 'button is-danger is-small', 'data-turbo-method' => :delete, data: { confirm: 'Are you sure?' } %>
```
Comment (alexisKinto) [inline app/views/documentation/documents/_list.html.erb:30]: Use the `Bali::DeleteLink::Component`

File: app/views/documentation/images/_grid.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<%= render Bali::ImageGrid::Component.new do |c| %>
+  <%= images.each do |image| %>
+    <%= c.image(id: dom_id(image), image_ratio: 'is-3by2') do |img| %>
+      <%= image_tag main_app.url_for(image.variant(:thumb)) %>
+      <% img.footer(class: 'p-2') do %>
+        <%= link_to render(Bali::Icon::Component.new('trash')), page_image_path(page, image), 
+                    class: 'button is-danger is-small', 'data-turbo-method' => :delete,
+                    data: { confirm: 'Are you sure?' } %>
+      <% end %>
```
Comment (alexisKinto) [inline app/views/documentation/images/_grid.html.erb:9]: Use `DeleteLink::Component`

File: app/controllers/documentation/documents_controller.rb (hunk 2 above)
Comment (rafaelcdelgado) [reply app/controllers/documentation/documents_controller.rb:29]: Why do we need to add an extra method?

File: app/models/documentation/page.rb (hunk 3 above)
Comment (rafaelcdelgado) [reply app/models/documentation/page.rb:61]: mmm, I Agree...

File: app/views/documentation/documents/_list.html.erb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<%= render(Bali::Table::Component.new) do |c| %>
+  <%= c.headers([
+          { name: t('documentation.document.name') },
+          { name: t('documentation.document.type') },
+          { name: t('documentation.document.uploaded_at') },
+          { name: '' }
+        ]) %>
+  <% c.no_records_notification do %>
+    <%= render Bali::Icon::Component.new('alert-alt') %>
+    <p>So sorry, no records found!</p>
```
Comment (alexisKinto) [inline app/views/documentation/documents/_list.html.erb:10]: Use translations

File: app/views/documentation/documents/_list.html.erb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        <%= page.file_extension(document) %>
+      </td>
+      <td>
+        <%= l(document.created_at, format: :date_twelve_hours) %>
+      </td>
+      <td>
+        <%= render(Bali::DeleteLink::Component.new(
+            href: page_document_path(@page, document),
+            name: render(Bali::Icon::Component.new('trash')),
+            confirm: 'Are you sure?')) %>
```
Comment (alexisKinto) [inline app/views/documentation/documents/_list.html.erb:32]: I think there is a default confirm message already

File: app/views/documentation/documents/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  <% c.tab(title: 'Documents', icon: 'file-signature') do %>
+    <%= turbo_frame_tag 'page_documents' do %>
+
+      <%= render 'form', page: @page %>
+
+      <hr />
+
+      <%= render 'documentation/pages/documents', documents: @documents, page: @page %>
+
+    <% end %>
```
Comment (alexisKinto) [inline app/views/documentation/documents/index.html.erb:11]:
```suggestion
    <%= turbo_frame_tag 'page_documents' do %>
      <%= render 'form', page: @page %>
      <hr />
      <%= render 'documentation/pages/documents', documents: @documents, page: @page %>
    <% end %>
```

Comment (alexisKinto) [reply app/views/documentation/documents/index.html.erb:11]: Same on files/index

File: spec/models/documentation/page_spec.rb [hunk 9]
```diff
@@ -80,4 +81,36 @@
...
+      comedor_recipes_details.documents.attach(document_file)
+
+      doc = comedor_recipes_details.documents.first
+
+      expect(comedor_recipes.file_extension(doc)).to eql('PDF')
+    end
+  end
+
+  describe '#allowed_extensions' do
+    context 'when file is document' do
```
Comment (alexisKinto) [inline spec/models/documentation/page_spec.rb:96]:
```suggestion
    context 'when file is a document' do
```

File: spec/models/documentation/page_spec.rb [hunk 10]
```diff
@@ -80,4 +81,36 @@
...
+          '.docx',
+          '.xls',
+          '.xlsx',
+          '.ppt',
+          '.pptx'
+        )
+      end
+    end
+
+    context 'when the file is image' do
```
Comment (alexisKinto) [inline spec/models/documentation/page_spec.rb:110]:
```suggestion
    context 'when the file is an image' do
```

File: spec/requests/documentation/documents_spec.rb [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+module Documentation
+  RSpec.describe 'Documents', type: :request do
+    include Engine.routes.url_helpers
+
+    fixtures 'documentation/pages'
+
+    let(:page) { documentation_pages(:comedor_home_page) }
+    let(:document_file) { fixture_file_upload('file.pdf', 'application/pdf') }
+
+    describe 'GET /index' do
```
Comment (alexisKinto) [inline spec/requests/documentation/documents_spec.rb:14]: I suggest showing the completed path

**Review by alexisKinto — APPROVED** [review]
Minor suggestions

File: spec/requests/documentation/documents_spec.rb (hunk 11 above)
Comment (rafaelcdelgado) [reply spec/requests/documentation/documents_spec.rb:14]: I just do it and it looks really ugly, I really prefer the action instead the full path, is really hard to read.

**Pull request merged**